docs/content/AutoGPT/component agent/components.md

```diff
@@ -0,0 +1,119 @@
+# 🧩 Components
+
+Components are the building blocks of [🤖 Agents](./agents.md). They are classes inheriting `AgentComponent` or implementing one or more [⚙️ Protocols](./protocols.md) that give agent additional abilities or processing. 
+
+By default components assigned to attributes (fields) in agent's `__init__` are automatically discovered upon instantiation.
+Each component can implement multiple protocols and can rely on other components if needed.
```

Comment:
Clarify this, it doesn't make sense to me. How do I know what is assigned to attributes?

Reply:
Does self.other_componet also work?

Reply:
Yes, the name doesn't matter, what matters is the type. My goal of this approach was to simplify building/extending an agent. Components are discovered automatically using metaclass on instantiation. If that isn't clear I can introduce `add_component` method but that's unnecessary.